src/core/qgsfieldformatter.cpp

```diff
@@ -30,13 +30,17 @@ QString QgsFieldFormatter::representValue( QgsVectorLayer *layer, int fieldIndex
   if ( layer->fields().fieldOrigin( fieldIndex ) == QgsFields::OriginProvider && layer->dataProvider() )
     defVal = layer->dataProvider()->defaultValueClause( layer->fields().fieldOriginIndex( fieldIndex ) );
 
-  if ( ! layer->fields().exists( fieldIndex ) )
+  if ( !defVal.isNull() && defVal == value )
```

Comment:
was the test ``! layer->fields().exists( fieldIndex )`` removed on purpose ?
I see that QgsFields::fieldOrigin() does internally a exists() test so the previous implementation of representValue() was robust to a invalid fieldIndex, and no longer is. Perhaps the documentation of the method should reflect that fieldIndex must be valid.

Reply:
Good catch. I restored the check.  And propose #45315 to fix it on master